src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParserStateObject.cs

```diff
@@ -20,6 +20,102 @@ sealed internal class LastIOTimer
         internal long _value;
     }
 
+    internal readonly ref struct PacketHandle
+    {
+        public const int NativePointerType = 1;
+        public const int NativePacketType = 2;
+        public const int ManagedPacketType = 3;
+
+#if FEATURE_INTEROPSNI
```

Comment:
We are trying to stay away from these `#if defs` in code as much as possible. In some cases the work is way too much to remove them. 

In this case, you should move the struct to a new file.
The file with code behind `#if FEATURE_INTEROPSNI` should be part of Windows compilation only. 

This should result in PacketHandle.Windows.cs and PacketHandle.Unix.cs

There may be duplicate code, but that is OK. Reading with If Def is much harder.

I haven't reviewed the rest of this Code. I wanted to put out this comment so that you can act on it.

Reply:
The common code could go into a partial struct which can be compiled on both Windows and Unix.

Reply:
Done. It isn't possible to use partial readonly structs with fields defined in multiple  units because there is no way to specify the field ordering, it also isn't possible to setup a constructor to be definitely defined in all cases which causes an compiler error. So I just made separate windows and unix versions and added them to the project in the same way that other such files are used in this project.